### PR TITLE
OHRM5X-785:  upgrader screens

### PR DESCRIPTION
**Pull request by Ajiharan007**
This PR contains 
OHRM5X-787: system check page
OHRM5X-1248:current version details screen
OHRM5X-1266: welcome page screen
OHRM5X-785: Database Configuration screen
OHRM5X-1255-upgrader complete screen

### CODE REVIEW

File: installer/client/src/pages/DatabaseConfigScreen.vue [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    <oxd-text tag="h4" class="orangehrm-database-info-title"
+      >Database Information</oxd-text
+    >
+    <oxd-text class="orangehrm-database-info-content"
+      >Please provide the database information of the database you are going to
+      upgrade.</oxd-text
+    >
+    <Notice title="important">
+      <oxd-text tag="p" class="orangehrm-current-version-content">
+        If yey.ohrmou have enabled data encrypted in your current version, you
```
Comment (RajithaKumara) [inline installer/client/src/pages/DatabaseConfigScreen.vue:31]: `yey.ohrmou`? :thinking:

File: installer/client/src/pages/WelcomeScreen.vue [hunk 2]
```diff
@@ -18,7 +18,7 @@
  */
  -->
 <template>
-  <oxd-form class="orangehrm-installer-page">
+  <oxd-form class="orangehrm-installer-page" @submit="showModel">
     <oxd-text tag="h5" class="orangehrm-installer-page-title">
       Welcome to OrangeHRM Stater Version 5.0 Setup Wizard
```
Comment (RajithaKumara) [inline installer/client/src/pages/WelcomeScreen.vue:23]:
```suggestion
      Welcome to OrangeHRM Stater Version {productVersion} Setup Wizard
```
will discuss this

File: installer/client/src/pages/WelcomeScreen.vue [hunk 3]
```diff
@@ -59,7 +59,12 @@
 
     <Notice title="important">
       <oxd-text tag="p" class="orangehrm-installer-page-content">
-        Test
+        OrangeHRM Starter 5.0 only supports Admin,PIM,Leave,Time and
+        Attendance,and Maintenance modules.if you are utilizing other modules pr
+        Add-ons and are thinking about upgrading, we recommended waiting for
+        version5.1.You can,however, continue with the update while ignoring the
+        unsupported modules and add-ons,but please note that data will be
+        inaccessible
```
Comment (RajithaKumara) [inline installer/client/src/pages/WelcomeScreen.vue:67]:
```suggestion
        inaccessible.
```

File: installer/client/src/pages/DatabaseConfigScreen.vue [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    };
+  },
+  methods: {
+    onSubmit() {
+      const {hostName, hostPort, databaseName, userName, userPassword} =
+        this.database;
+      this.http.request({
+        method: 'post',
+        data: {hostName, hostPort, databaseName, userName, userPassword},
+      });
```
[the remark this answers is not in the document]
Comment (Ajiharan007) [reply installer/client/src/pages/DatabaseConfigScreen.vue:164]: not yet

File: installer/client/src/pages/WelcomeScreen.vue (hunk 2 above)
Comment (Ajiharan007) [reply installer/client/src/pages/WelcomeScreen.vue]: can i use API or prop?

File: installer/client/src/pages/CurrentVersionScreen.vue [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    </oxd-text>
+
+    <oxd-form-row class="orangehrm-current-version-row">
+      <oxd-grid :cols="3" class="orangehrm-full-width-grid">
+        <oxd-grid-item>
+          <oxd-input
+            v-model="config.version"
+            type="select"
+            label="Current OrangeHRM Version"
+            :options="getItems"
```
Comment (Super-Chama) [inline installer/client/src/pages/CurrentVersionScreen.vue:42]: Computed property is not a function, its a property so **get**Items name is bad and computed property should be used for reactive purposes and transformations. here the versions are loaded only once from BE why do you need a computed property for it? 

```
getItems() {
      if (this.items.length == 0) {
        return [];
      }
      return JSON.parse(JSON.stringify(this.items)); // why ?
    },
```
The flow should be.

> 1.  Declare a state variable for version items
> 2.  Before component load, send a http request to get verison items from BE
> 3.  Assign fetched data to state variable
> 4.  Pass the state variable to dropdown component as props

Check how dropdown options are loaded in other places of app. 
https://github.com/orangehrm/orangehrm/blob/5.0/symfony/client/src/orangehrmPimPlugin/components/JobtitleDropdown.vue

File: installer/client/src/pages/CurrentVersionScreen.vue [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    color: $oxd-primary-one-color;
+    padding: 0 0.75rem 0.75rem 0.75rem;
+  }
+  &-content {
+    font-size: $oxd-input-control-font-size;
+    padding: 0.75rem;
+  }
+  ::v-deep(.oxd-grid-3) {
+    width: 100%;
+    margin: 0 !important;
```
Comment (Super-Chama) [inline installer/client/src/pages/CurrentVersionScreen.vue:136]: is this important needed? why do we set these for oxd-grid-3 ?

File: installer/client/src/pages/DatabaseConfigScreen.vue [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+<template>
+  <oxd-form class="orangehrm-database-info" @submit-valid="onSubmit">
+    <oxd-text tag="h5" class="orangehrm-database-info-title">
+      Database Information
+    </oxd-text>
+    <oxd-text class="orangehrm-database-info-content"
+      >Please provide the database information of the database you are going to
```
Comment (Super-Chama) [inline installer/client/src/pages/DatabaseConfigScreen.vue:30]: manual format

File: installer/client/src/pages/SystemCheckScreen.vue [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+<template>
+  <oxd-form :loading="isLoading">
+    <div class="orangehrm-system-check">
+      <oxd-text
+        tag="h5"
+        class="orangehrm-system-check-content orangehrm-system-check-title"
+        >System Check</oxd-text
```
Comment (Super-Chama) [inline installer/client/src/pages/SystemCheckScreen.vue:26]: Manual format

File: installer/client/src/pages/SystemCheckScreen.vue [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+        <oxd-button display-type="ghost" label="Back" @click="navigateUrl" />
+        <oxd-button
+          display-type="ghost"
+          label="Re-Check"
+          type="submit"
+          @click="Recheck"
+        />
+        <oxd-button
+          display-type="ghost"
+          :disabled="checkErrors"
```
Comment (Super-Chama) [inline installer/client/src/pages/SystemCheckScreen.vue:54]: see my comment about computed properties above. `checkErrors` should be `hasErrors`

File: installer/client/src/pages/SystemCheckScreen.vue [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+      });
+  },
+  methods: {
+    navigateUrl() {
+      navigate('/upgrader/database-config');
+    },
+    goToScreen() {
+      navigate('/upgrader/current-version');
+    },
+    Recheck() {
```
Comment (Super-Chama) [inline installer/client/src/pages/SystemCheckScreen.vue:123]: Recheck should be camelCase

File: installer/client/src/pages/WelcomeScreen.vue [hunk 11]
```diff
@@ -71,23 +76,48 @@
...
+    version: {
+      type: String,
+      required: true,
+    },
   },
   data() {
     return {
-      selected: "install",
+      selected: 'install',
+      hasShowModel: false,
```
Comment (Super-Chama) [inline installer/client/src/pages/WelcomeScreen.vue:108]: showModal

**Review by Super-Chama — CHANGES_REQUESTED** [review]

File: installer/client/src/pages/CurrentVersionScreen.vue (hunk 6 above)
Comment (Ajiharan007) [reply installer/client/src/pages/CurrentVersionScreen.vue:136]: important is not needed, but need to override width & margin

File: installer/Controller/CurrentVersionDetailsController.php [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+{
+    /**
+     * @inheritDoc
+     */
+    public function preRender(Request $request): void
+    {
+        $component = new Component('current-version-screen');
+        $this->setComponent($component);
+    }
+}
```
[the remark this answers is not in the document]
Comment (Super-Chama) [reply installer/Controller/CurrentVersionDetailsController.php:35]: @Ajiharan007 you have marked as resolved yet the line ending is still not there. please only resolve after its fixed
image (6)

File: installer/client/src/pages/WelcomeScreen.vue [hunk 13]
```diff
@@ -71,23 +79,52 @@
...
+  methods: {
+    toggleModal() {
+      this.showModal = !this.showModal;
+    },
+    closeModel(isAccept) {
+      if (!isAccept) return this.toggleModal();
+      if (this.selected === 'upgrade') {
+        navigate('/upgrader/database-config');
+      } else {
+        window.location.href = window.location.pathname.replace(
```
Comment (Super-Chama) [inline installer/client/src/pages/WelcomeScreen.vue:121]: will check this

File: installer/client/src/styles/_variables.scss [hunk 14]
```diff
@@ -19,4 +19,5 @@ $oxd-font-family: 'Nunito Sans' !default;
 /* Import OXD Globals */
 @import '@ohrm/oxd/styles/_colors.scss';
 @import '@ohrm/oxd/styles/_variables.scss';
+@import '@ohrm/oxd/core/components/Alert/variables.scss';
```
Comment (Super-Chama) [inline installer/client/src/styles/_variables.scss:22]: why do you need this globally imported?

**Review by Super-Chama — APPROVED** [review]
I approve these changes with the condition that you will check with BA and fix spelling and grammar mistakes.

File: installer/client/src/styles/_variables.scss (hunk 14 above)
Comment (Ajiharan007) [reply installer/client/src/styles/_variables.scss:22]: other places may be needed

File: installer/client/src/core/util/helper/password.ts [hunk 15]
```diff
@@ -0,0 +1,25 @@
+export function getPassLevel(password: string): number[] {
```
Comment (RajithaKumara) [inline installer/client/src/core/util/helper/password.ts:1]: missing copyright header

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**